engine/src/keysinuse_ec.c

```diff
@@ -51,27 +51,30 @@ static void ec_index_free_key(void *parent, void *ptr, CRYPTO_EX_DATA *ad,
     EC_KEY *eckey = (EC_KEY *)parent;
     keysinuse_info *info = (keysinuse_info *)ptr;
 
-    if (!global_logging_disabled() &&
-        info != NULL &&
-        (info->encrypts > 0 || info->decrypts > 0))
+    if (info != NULL)
     {
-        if (info->key_identifier[0] == '\0' &&
-            !get_ec_key_identifier(eckey, info))
+        if (!global_logging_disabled() &&
+            (info->encrypts > 0 || info->decrypts > 0))
         {
-            return;
+            if (info->key_identifier[0] == '\0' &&
+                !get_ec_key_identifier(eckey, info))
+            {
+                return;
```

Comment:
Isn't there still a potential for a leak with this return?

Reply:
Good catch. That should be the only potential leak left.